tests/js/pv-tests.js

```diff
@@ -0,0 +1,245 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+"use strict";
+
+/* global navigator, WorkerNavigator */
+
+const require_method = require;
+
+const Realm = require("realm");
+const TestCase = require("./asserts");
+const AppConfig = require("./support/testConfig");
+const { ObjectId, UUID } = Realm.BSON;
+
+const PvIntDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId",
+        breed: "string?",
+        name: "string",
+        realm_id: "int?"
+    }
+};
+
+const PvStringDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId",
+        breed: "string?",
+        name: "string",
+        realm_id: "string?"
+    }
+};
+
+const PvUuidDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId",
+        breed: "string?",
+        name: "string",
+        realm_id: "uuid?"
+    }
+};
+
+const PvObjectIdDog = {
+    name: "Dog",
+    primaryKey: "_id",
+    properties: {
+        _id: "objectId",
+        breed: "string?",
+        name: "string",
+        realm_id: "objectId?"
+    }
+};
+
+const createConfig = (schema, user, partitionValue) => ({
+    schema: Array.isArray(schema) ? schema : [schema],
+    sync: {
+        user,
+        partitionValue,
+        _sessionStopPolicy: "immediately", // Make it safe to delete files after realm.close()
+    }
+});
+
+module.exports = {
+    async testPartitionValueAsInt() {
```

Comment:
Since these tests are simply variations on the input and output, I'm wondering if it would be simpler to maintain if they all called a shared function? Or if they were simply built from an array of inputs (schema, properties of the inserted object, etc.)

Reply:
Since our tests are up for a refactor, I would like to leave it for that. But I agree, and the thought did cross my mind.